libs/cardano-ledger-core/src/Cardano/Ledger/Core.hs

```diff
@@ -648,3 +619,43 @@ wireSizeTxF =
         then fromIntegral n
         else error $ "Impossible: Size of the transaction is too big: " ++ show n
 {-# INLINEABLE wireSizeTxF #-}
+
+-- | Translate a transaction through its binary representation from previous to current era.
+binaryUpgradeTx ::
+  forall era.
+  (Era era, ToCBOR (Tx (PreviousEra era)), DecCBOR (Annotator (Tx era))) =>
+  -- | Label for error reporting
+  Text ->
+  Tx (PreviousEra era) ->
+  Except DecoderError (Tx era)
+binaryUpgradeTx msg tx = translateViaCBORAnnotator (eraProtVerLow @era) msg tx
+
```

Comment:
I don't think we need to supply custom message, since result is `Either` and custom message can always be create by the user later.
```suggestion
  Tx (PreviousEra era) ->
  Except DecoderError (Tx era)
binaryUpgradeTx tx = translateViaCBORAnnotator (eraProtVerLow @era) "Tx" tx

```
If you'd like, you could also report an era with a helper:
```suggestion
  Tx (PreviousEra era) ->
  Except DecoderError (Tx era)
binaryUpgradeTx tx = translateViaCBORAnnotator (eraProtVerLow @era) (withEraName @era "Tx") tx

withEraName :: forall era. Era era => Text -> Text
withEraName t = t <> " " <> T.pack (eraName @era) <> "Era"

```

Reply:
Nice, thank you!